playlist_souffle/use_cases/souffle_playlist.py

```diff
@@ -0,0 +1,36 @@
+"""Module for the souffle_playlist use case."""
+
+from datetime import datetime as dt
+from playlist_souffle.definitions.playlist import Playlist
+import playlist_souffle.entities.souffler as souffler_entity
+import playlist_souffle.entities.track as track_entity
+
+def souffle_playlist(spotify, playlist_uri, user_id, shuffle_by):
+    """Souffle a playlist."""
+
+    original_playlist = Playlist(
+        owner=None,
+        name=spotify.fetch_playlist_name(playlist_uri),
+        tracks=spotify.fetch_playlist_tracks(playlist_uri),
+        description=None
+    )
+
+    collection_id_by_track = {track: track_entity.extract_track_collection_id(track, shuffle_by)
+                              for track in original_playlist.tracks}
+
+    collection_tracks_by_track = spotify.fetch_collection_tracks_by_track(
+        collection_id_by_track,
```

Comment:
seems like there's some duplicate info here. if you have the track and it's collection id, i guess you also need the type of collection it is?

Reply:
Just want to float the idea that track_collection is a thing you might want to make into an entity. you have three types: playlist, album, artist. (this might be a terrible idea.)

Reply:
maybe they are called `related_tracks` and that has: tracks, relation_type, relation_id? 

agreed about the whole playlist being a separate thing though. nice way of describing it.

Reply:
Hmm i like this but think relation_id is not super clear.

```RelatedTracks = namedtuple('RelatedTracks', 'tracks relation_type ?relation_id?')```

I really like the function being called
```fetch_related_tracks_by_track``` but remember (just realized this) we'd still have to pass the relation type and id since we wont actually have the tracks yet. Which means same issue with duplication.

Reply:
👍 on `related_by`.

Reply:
I like how this function reads in the latest commit